locator/network_topology_strategy.cc

```diff
@@ -336,39 +335,128 @@ future<tablet_map> network_topology_strategy::allocate_tablets_for_new_table(sch
         tablet_count = aligned_tablet_count;
     }
 
-    tablet_map tablets(tablet_count);
+    return reallocate_tablets(std::move(s), std::move(tm), tablet_count, nullptr);
+}
+
+future<tablet_map> network_topology_strategy::reallocate_tablets(schema_ptr s, token_metadata_ptr tm, size_t tablet_count, const tablet_map* cur_tablets) const {
```

Comment:
THe cur_tablets is non-null for tests only, why not make _tests_ pass here the `tablet_map(tablet_count)`?

Reply:
The idea is for RF changes to use this interface and passing a pointer to the current tablet_map.
I'm not sure what exactly you're suggesting.

Reply:
good idea

Reply:
> The idea is for RF changes to use this interface and passing a pointer to the current tablet_map.

Ah, OK. Then my comment is irrelevant

Reply:
done in v2